packages/calypso-products/src/get-price-tier-for-units.ts

```diff
@@ -2,8 +2,11 @@ export interface PriceTierEntry {
 	minimum_units: number;
 	maximum_units?: undefined | null | number;
 	minimum_price: number;
+	minimum_price_in_display_units: number;
 	minimum_price_display: string;
-	minimum_price_monthly_display: string;
+	minimum_price_monthly_display?: string | null | undefined;
+	maximum_price_in_display_units: string;
```

Comment:
I think that this name doesn't properly suggest what this value holds. What is a "display unit"? `minimum_price_monthly_display` means "the minimum price formatted for the local currency with the appropriate currency symbol".

Reply:
Any suggestions for how to name this ? 

What we want `maximum_price_in_display_units` to mean is the minimum_price in "non small units". **Ideally** minimum_price would have had minimum_price_smallest_unit. But it just has minimum_price.

Earlier me and @aneeshd16 had a discussion here and we couldn't reach consensus on the following variations. 😅 
- maximum_price_decimal_units
- maximum_price_ normal_units
- maximum_price_large_units
- maximum_price_non_smallest_units

We also could not come up with a better name for this 😬

Reply:
I understand. Naming is hard! By "non small units", you mean like, "the way in which the currency is usually displayed"? Like, USD is a decimal with dollars and cents, and JPY is a non-decimal. 

My suggestion, if it works for you, is to get the integer value of the price in the currency's smallest unit (typically we've used names like `..._integer` for this, like [total_cost_integer](https://github.com/Automattic/wp-calypso/blob/trunk/packages/shopping-cart/src/types.ts#L260), and then call the [formatCurrency](https://github.com/Automattic/wp-calypso/tree/972be183f14ad4d6a7a27f3d4eb3936128ab7398/packages/format-currency) function with `isSmallestUnit: true` as the option.

Here's some examples:

https://github.com/Automattic/wp-calypso/blob/17d1b9d9b1c4f2cf24183426cb2628073cdc14d9/packages/wpcom-checkout/src/checkout-line-items.tsx#L659-L666

https://github.com/Automattic/wp-calypso/blob/17d1b9d9b1c4f2cf24183426cb2628073cdc14d9/client/lib/purchases/index.ts#L786-L789

Reply:
Yup we had this discussion actually with @aneeshd16  ( p1662968063004569/1662960732.084959-slack-C02KVCAL7GX ). The problem is I need to "Sum" two values. One of these values ( Price tier values ) are in the smallest units and the other value ( [ price on the productList ](https://github.com/Automattic/wp-calypso/blob/01ab2e6795c908591ee174f2587a6cef0934c79a/client/state/products-list/selectors/get-products-list.ts#L25) ) are in "the way in which the currency is usually displayed" and not in small units. 

So it is necessary to get both the price tier values and product list values in the same type of unit. That is why I need to get these values. So that I can first sum and the format the currency

Reply:
That makes sense. It looks like maybe you figured it out by adding `cost_smallest_unit`, which sounds to me like the right solution. The safest normalized unit to use for any math is the smallest unit in the currency, so that there's no risk of decimal math errors. 👍